app/models/CustomerInformation.scala

```diff
@@ -52,23 +54,72 @@ case class CustomerInformation(organisationName: Option[String],
+      partyType <- (__ \ "partyType").readNullable[String].orElse(Reads.pure(None))
+      primaryMainCode <- (__ \ "primaryMainCode").read[String]
+      stdReturnPeriod <- (__ \\ "stdReturnPeriod").readNullable[String].orElse(Reads.pure(None))
+      nonStdReturnPeriod <- (__ \\ "nonStdTaxPeriods").readNullable[Seq[TaxPeriod]].orElse(Reads.pure(None))
+      firstNonNSTPeriod <- (__ \\ "firstNonNSTPPeriod").readNullable[TaxPeriod].orElse(Reads.pure(None))
+      mandationStatus <- (__ \ "pendingChanges" \ "mandationStatus").readNullable[String].orElse(Reads.pure(None))
+      deregistration <- (__ \ "deregistration").readNullable[Deregistration].orElse(Reads.pure(None))
+      changeIndicators <- (__ \ "changeIndicators").readNullable[ChangeIndicators].orElse(Reads.pure(None))
+      missingTrader <- (__ \ "missingTrader").read[Boolean]
+      hasPendingPpobChanges <- (__ \ "pendingChanges" \ "ppob").readNullable[JsObject].orElse(Reads.pure(None)).map(_.nonEmpty)
+    } yield {
+      CustomerInformation(
+        organisationName,
+        firstName,
+        lastName,
+        tradingName,
+        ppob,
+        contactDetails,
+        isPartialMigration,
+        customerMigratedToEtmpDate,
+        vatRegistrationDate,
+        partyType,
+        primaryMainCode,
+        stdReturnPeriod,
+        nonStdReturnPeriod,
+        firstNonNSTPeriod,
+        mandationStatus,
+        deregistration,
+        changeIndicators,
+        missingTrader,
+        hasPendingPpobChanges
+      )
+    }
+  }
+
+
+
+//    (
+//    (JsPath \ "customerDetails" \ "organisationName").readNullable[String].orElse(Reads.pure(None)) and
+//    (JsPath \ "customerDetails" \ "firstName").readNullable[String].orElse(Reads.pure(None)) and
+//    (JsPath \ "customerDetails" \ "lastName").readNullable[String].orElse(Reads.pure(None)) and
+//    (JsPath \ "customerDetails" \ "tradingName").readNullable[String].orElse(Reads.pure(None)) and
+//    (JsPath \ "ppob").read[Address] and
+//    (JsPath \ "ppob" \ "contactDetails").readNullable[Email].orElse(Reads.pure(None)) and
+//    (JsPath \\ "isPartialMigration").readNullable[Boolean].map(_.contains(true)) and
+//    (JsPath \\ "customerMigratedToETMPDate").readNullable[String] and
+//    (JsPath \\ "vatRegistrationDate").readNullable[String] and
+//    (JsPath \ "partyType").readNullable[String] and
+//    (JsPath \ "primaryMainCode").read[String] and
+//    (JsPath \\ "stdReturnPeriod").readNullable[String] and
+//    (JsPath \\ "nonStdTaxPeriods").readNullable[Seq[TaxPeriod]] and
+//    (JsPath \\ "firstNonNSTPPeriod").readNullable[TaxPeriod] and
+//    (JsPath \ "pendingChanges" \ "mandationStatus").readNullable[String].orElse(Reads.pure(None)) and
+//    (JsPath \ "deregistration").readNullable[Deregistration] and
+//    (JsPath \ "changeIndicators").readNullable[ChangeIndicators] and
+//    (JsPath \ "missingTrader").read[Boolean]
+//  )(CustomerInformation.apply _)
```

Comment:
Are all these commented lines supposed to still be here?

Reply:
No. We're never supposed to have comments. MASSIVE derp on my part. Herpaderp